tests/unit/remote/test_http.py

```diff
@@ -13,3 +21,19 @@ def test_no_traverse_compatibility(dvc_repo):
 
     with pytest.raises(ConfigError):
         RemoteHTTP(dvc_repo, config)
+
+
+def test_download_fails_on_error_code(dvc_repo):
+    class ErrorStatusRequestHandler(BaseHTTPRequestHandler):
+        def do_GET(self):
+            self.send_response(404, message="Not found")
+            self.end_headers()
```

Comment:
Do we actually need it? Can we simply ask for some missing url?

Reply:
Well, 404 is quite easy to generate

Reply:
The question now is, should we leave changes made to StaticFileServer.
@Suor what do you think? I think they should stay, as the previous version was mapping string to its handler class, which was unnecessary.

Reply:
It's cleaner the new way.